Modules/ZDC/src/ZDCRawDataCheck.cxx

```diff
@@ -31,30 +36,49 @@ namespace o2::quality_control_modules::zdc
 
 Quality ZDCRawDataCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
 {
-  Quality result = Quality::Null;
 
+  Quality result = Quality::Null;
+  init();
+  mNumE = 0;
+  mNumW = 0;
+  mStringW = "";
+  mStringE = "";
+  int ib = 0;
   for (auto& [moName, mo] : *moMap) {
 
     (void)moName;
-    if (mo->getName() == "example") {
+    if (mo->getName() == "hpedSummary") {
       auto* h = dynamic_cast<TH1F*>(mo->getObject());
-
-      result = Quality::Good;
-
+      // dumpVecParam((int)h->GetNbinsX(),(int)mVectParam.size());
+      if ((int)h->GetNbinsX() != (int)mVectParam.size())
+        return Quality::Null;
       for (int i = 0; i < h->GetNbinsX(); i++) {
-        if (i > 0 && i < 8 && h->GetBinContent(i) == 0) {
-          result = Quality::Bad;
-          result.addReason(FlagReasonFactory::Unknown(),
-                           "It is bad because there is nothing in bin " + std::to_string(i));
-          break;
-        } else if ((i == 0 || i > 7) && h->GetBinContent(i) > 0) {
-          result = Quality::Medium;
-          result.addReason(FlagReasonFactory::Unknown(),
-                           "It is medium because bin " + std::to_string(i) + " is not empty");
-          result.addReason(FlagReasonFactory::BadTracking(),
-                           "This is to demonstrate that we can assign more than one Reason to a Quality");
+        ib = i + 1;
+        if ((((float)h->GetBinContent(ib) < (float)mVectParam.at(i).minW && (float)h->GetBinContent(ib) >= (float)mVectParam.at(i).minE)) || ((float)h->GetBinContent(ib) > (float)mVectParam.at(i).maxW && (float)h->GetBinContent(ib) < (float)mVectParam.at(i).maxE)) {
+          mNumW += 1;
+          mStringW = mStringW + mVectParam.at(i).ch + " ";
+          ILOG(Warning, Support) << "Raw Warning in " << mVectParam.at(i).ch << " intervall: " << mVectParam.at(i).minW << " - " << mVectParam.at(i).maxW << " Value: " << h->GetBinContent(ib) << ENDM;
+        }
+        if (((float)h->GetBinContent(ib) < (float)mVectParam.at(i).minE) || ((float)h->GetBinContent(ib) > (float)mVectParam.at(i).maxE)) {
+          mNumE += 1;
+          mStringE = mStringE + mVectParam.at(i).ch + " ";
+          ILOG(Error, Support) << "Raw Error in " << mVectParam.at(i).ch << " intervall: " << mVectParam.at(i).minE << " - " << mVectParam.at(i).maxE << " Value: " << h->GetBinContent(ib) << ENDM;
         }
       }
+      if (mNumW == 0 && mNumE == 0) {
+        result = Quality::Good;
+        result.addReason(FlagReasonFactory::Unknown(), "It is Good");
```

Comment:
If it is good, there is no need to add a `FlagReason` which indicates that the quality is bad. I would remove it.